packages/trie/src/checkpointTrie.ts

```diff
@@ -71,15 +44,13 @@ export class CheckpointTrie extends BaseTrie {
    * parent checkpoint as current.
    */
   async revert(): Promise<void> {
+    if (!this.isCheckpoint) {
+      throw new Error('trying to revert when not checkpointed')
+    }
+
     await this.lock.wait()
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
can this `no-unnecessary-condition` line be removed?

Reply:
Yes, have removed.